### PR TITLE
Next rc

### PR DESCRIPTION
**Pull request by hardiesoft**
Let's merge this into main now

### CODE REVIEW

File: _release/cacophony-api-report-stopped-devices [hunk 1]
```diff
@@ -1,4 +1,4 @@
 #
 # cron.d/cacophony-api-stopped-devices
-# every day at 11
-0 11 * * * root (cd /srv/cacophony/api/scripts/ && node --no-warnings=ExperimentalWarnings --experimental-json-modules report-stopped-devices.js) 2>&1 | logger --tag report-errors
+# every hour
```
Comment (gferraro) [inline _release/cacophony-api-report-stopped-devices:3]: thanks

File: api/api/V1/Device.ts [hunk 2]
```diff
@@ -444,6 +451,104 @@ export default function (app: Application, baseUrl: string) {
...
+   * @api {post} /api/v1/devices/remove-schedule Remove a schedule from a device.
+   * @apiName RemoveScheduleFromDevice
+   * @apiGroup Schedules
+   * @apiDescription This call removes a schedule from a device.
+   *
+   * @apiUse V1UserAuthorizationHeader
+   *
+   * @apiBody {Number} deviceId ID of the device.
+   * @apiBody {Number} scheduleId ID of the schedule to remove from the device.
+   * @apiBody {Boolean} admin If true, the user should have administrator access to the device.
```
Comment (gferraro) [inline api/api/V1/Device.ts:513]: is this used?

File: api/api/V1/Device.ts [hunk 3]
```diff
@@ -444,6 +451,104 @@ export default function (app: Application, baseUrl: string) {
...
+   * @apiQuery {Boolean} [only-active=true] Only operate if the device is active
```
Comment (gferraro) [inline api/api/V1/Device.ts:514]: should this be only-active=false
Same for other endpoint

File: api/api/V1/File.ts [hunk 4]
```diff
@@ -73,38 +90,24 @@ export default (app: Application, baseUrl: string) => {
...
-      if (request.query.offset == null) {
-        request.query.offset = "0";
-      }
-
-      if (request.query.limit == null) {
-        request.query.limit = "100";
-      }
-
+    extractJwtAuthorizedUser,
+    validateFields([query("type").equals("audioBait")]),
```
Comment (gferraro) [inline api/api/V1/File.ts:94]: missing from api docs
And are we no longer using BaseQueryParams?

File: api/api/V1/Group.ts [hunk 5]
```diff
@@ -330,6 +340,32 @@ export default function (app: Application, baseUrl: string) {
     }
   );
 
+  /**
+   * @api {get} api/v1/groups/:groupIdOrName/schedules Get audio bait schedules for a group
+   * @apiName GetSchedulesForGroup
+   * @apiGroup Schedules
+   * @apiDescription This call is used to retrieve the any audio bait schedules for a group.
+   * @apiUse V1UserAuthorizationHeader
+   *
```
Comment (gferraro) [inline api/api/V1/Group.ts:349]: parameter description missing

File: api/api/V1/Tag.ts [hunk 6]
```diff
@@ -93,15 +96,17 @@ export default function (app: Application, baseUrl: string) {
...
-   * @apiuse V1ResponseError
+   * @apiUse V1ResponseError
    *
    */
   app.delete(
     apiUrl,
     extractJwtAuthorizedUser,
     validateFields([idOf(body("tagId"))]),
+
+    // FIXME - Get the recording, if the user has access to it then we're good.
```
Comment (gferraro) [inline api/api/V1/Tag.ts:107]: hmm can anyone delete recording tags at the moment? seems dodgy

**Review by gferraro — APPROVED** [review]
Looks good, few api doc stuff that needs to be corrected

File: api/api/V1/Tag.ts (hunk 6 above)
Comment (hardiesoft) [reply api/api/V1/Tag.ts:107]: Nah, it's checked in Tag.deleteFromId, I just want to move that kind of permissions checking to the middleware for consistency.

**Pull request merged**